test/AtlasVerification.t.sol

```diff
@@ -0,0 +1,1369 @@
+    //     where the solver op contract is the atlas contract
+    // when validCalls is called from the userEOA
+    // then it should return NoSolverOp
+    // because the solver op contract must not be atlas
+    //
+    function test_validCalls_SolverContractIsAtlas_NoSolverOp() public {
+        defaultAtlasEnvironment();
+
+        UserOperation memory userOp = validUserOperation().build();
+        SolverOperation[] memory solverOps = new SolverOperation[](1);
+        SolverOperation memory solverOp = validSolverOperation(userOp).withSolver(address(atlas)).signAndBuild(address(atlasVerification), solverOnePK);
+        solverOps[0] = solverOp;
+        DAppOperation memory dappOp = validDAppOperation(userOp, solverOps).build();
+
+        callAndAssert(ValidCallsCall({
+            userOp: userOp, solverOps: solverOps, dAppOp: dappOp, msgValue: 0, msgSender: userEOA, isSimulation: false}
+        ), ValidCallsResult.NoSolverOp);
+    }
+
+    //
+    // given a default atlas environment
+    //   and otherwise valid user, solver and dapp operations
+    //     where the solver op userOpHash is not the same as the user op userOpHash
+    // when validCalls is called from the userEOA
+    // then it should return NoSolverOp
+    // because the solver op userOpHash must be the same as the user op userOpHash
+    //
+    function test_validCalls_SolverOpUserOpHashWrong_NoSolverOp() public {
+        defaultAtlasEnvironment();
+
+        UserOperation memory userOp = validUserOperation().build();
+        SolverOperation[] memory solverOps = new SolverOperation[](1);
+        SolverOperation memory solverOp = validSolverOperation(userOp).withUserOpHash(bytes32(0)).signAndBuild(address(atlasVerification), solverOnePK);
+        solverOps[0] = solverOp;
+        DAppOperation memory dappOp = validDAppOperation(userOp, solverOps).build();
+
+        callAndAssert(ValidCallsCall({
+            userOp: userOp, solverOps: solverOps, dAppOp: dappOp, msgValue: 0, msgSender: userEOA, isSimulation: false}
+        ), ValidCallsResult.NoSolverOp);
+    }
+
+    //
+    // given a default atlas environment
+    //   and otherwise valid user, solver and dapp operations
+    //     where there are no solver ops
+    // when validCalls is called from the userEOA
+    //   and isSimulation = true
+    // then it should return Valid
+    // because no solver ops is valid for simulations
+    //
+    function test_validCalls_NoSolverOps_Simulated_Valid() public {
+        defaultAtlasEnvironment();
+
+        UserOperation memory userOp = validUserOperation().build();
+        SolverOperation[] memory solverOps = new SolverOperation[](0);
+        DAppOperation memory dappOp = validDAppOperation(userOp, solverOps).build();
+
+        callAndAssert(ValidCallsCall({
+            userOp: userOp, solverOps: solverOps, dAppOp: dappOp, msgValue: 0, msgSender: userEOA, isSimulation: false}
+        ), ValidCallsResult.NoSolverOp);
```

Comment:
```suggestion
            userOp: userOp, solverOps: solverOps, dAppOp: dappOp, msgValue: 0, msgSender: userEOA, isSimulation: true}
        ), ValidCallsResult.Valid);
```
Fix settings here to match test description. Still passes on my end with the new settings.